_meta/ecs-migration.yml

```diff
@@ -35,42 +38,46 @@
 - from: context.system.platform
   to: host.os.platform
 
-#- from: context.request.method
-#  to: http.request.method
-#
-#- from: context.request.http_version
-#  to: http.version
-#
-#- from: context.request.body
-#  to: http.request.body
-#  index: false
-#
-## Not in ECS
-#- from: context.request.cookies
-#  to: http.request.headers.cookies.parsed
-#  index: false
-#
-## Not in ECS
-#- from: context.request.headers.cookie
-#  to: http.request.headers.cookies.original
-#  index: false
-#
-#- from: context.request.headers.user-agent
-#  to: http.request.headers.user-agent
-#  index: false
-#
-#- from: context.request.headers.content-type
-#  to: http.request.headers.content-type
-#  index: false
-#
-#- from: context.request.env
-#  to: http.request.env
-#  index: false
-#
-#- from: context.request.socket
-#  to: http.request.socket
-#  index: false
-#
+- from: context.request.method
+  to: http.request.method
+
+- from: context.request.http_version
+  to: http.version
+
+- from: context.request.body
+  to: http.request.body.content
+  index: false
+
+  # Not in ECS
+- from: context.request.cookies
+  to: http.request.headers.cookies.parsed
+  index: false
+
+  # Not in ECS
+- from: context.request.headers.cookie
+  to: http.request.headers.cookies.original
+  index: false
+
+  # Not in ECS
+- from: context.request.headers.user-agent
+  to: http.request.headers.user-agent.original
```

Comment:
Actually `user_agent` is in ECS: https://github.com/elastic/ecs#user_agent

Reply:
what is the reason for that? the actual header name is User-Agent

Reply:
I think this should stay as `user-agent` since it's nested under `http.request.headers` but I'd also be ok to switch if everyone else prefers that

Reply:
In ECS it's at the top level, so not really a conflict with how you normalize header names as field names (if you decide to move it at the top level).

If you decide to leave it nested here because you prefer to list it as one of the header entries, then I'd opt to leave as is, to keep the consistent usage of '-' with other headers.

My actual suggestion would be to move this to ECS, however (and I thought I had seen another entry in this file to do so).

Moving this field to the ECS name will ensure the user agents recorded by APM can be found easily by any integration or person that supports the common schema, without having to create an exception for APM. That's the whole point of ECS, after all :-)

Reply:
Yes, as you noted there is another entry in this file for that. We have 2 user agents: the one in the request to the apm-server (which is moved to the ECS field at the root) and this one, which is in a request possibly captured by an APM agent.

Reply:
Are these present in the same event at the same time? If not, then both should break down the user agent in the ECS field, `user_agent` at the top level.

If they can both be present in the same event, then that's a more complicated question :-)

Reply:
Yes, they are present at the same time.